src/main/java/net/virtualinfinity/nio/ConnectionInitiator.java

```diff
@@ -4,7 +4,9 @@
 import java.net.InetSocketAddress;
 import java.net.SocketAddress;
 import java.net.UnknownHostException;
-import java.nio.channels.*;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.ServerSocketChannel;
 import java.util.function.Consumer;
```

Comment:
Consumer and other java.util.function.* interfaces are Java 1.8, so you haven't really made it work with 1.7.  You could create an equivalent interface, this was just convenient to use in 1.8.